lib/utils/HttpClient.js

```diff
@@ -64,15 +82,34 @@ class HttpClient {
       .build();
   }
 
-  getCommand(url, method) {
-    url = _.toUpper(url);
-    method = _.toUpper(method);
-    if (this.commandMap[`${url}_${method}_CIRCUIT`]) {
-      return this.commandMap[`${url}_${method}_CIRCUIT`];
-    } else if (this.commandMap[`${method}_CIRCUIT`]) {
-      return this.commandMap[`${method}_CIRCUIT`];
-    } else if (this.commandMap.BASE_CIRCUIT) {
-      return this.commandMap.BASE_CIRCUIT;
+  getCommand(url, httpMethod) {
+    url = _.toLower(url);
+    httpMethod = _.toLower(httpMethod);
+    let baseUrl = this.baseUrl;
+    let path = url;
+    if (this.baseUrl === undefined || url.indexOf('http') !== -1) {
+      let urlComponents;
+      if (this.PARSED_URL_MAP[url] !== undefined) {
+        urlComponents = this.PARSED_URL_MAP[url];
+      } else {
+        urlComponents = parseUrl(url);
+        this.PARSED_URL_MAP[url] = urlComponents;
+      }
+      baseUrl = `${urlComponents.protocol}//${urlComponents.hostname}:${urlComponents.port}`;
+      path = urlComponents.pathname;
+      if (this.commandMap[baseUrl] === undefined) {
+        logger.silly('setting up command factory for url :-', baseUrl);
+        this.buildCommandFactory(baseUrl);
+      }
+    }
+    if (this.commandMap[baseUrl]) {
+      if (this.commandMap[baseUrl][`${baseUrl}_${httpMethod}_${path}_CIRCUIT`]) {
```

Comment:
will checking index with captialized `CIRCUIT` work here and below since we create the command map using lower case indexes in buildCommandFactory ?